lib/chef/cookbook/gem_installer.rb

```diff
@@ -45,10 +46,16 @@ def install
 
         unless cookbook_gems.empty?
           begin
-            inline_gemfile do
-              source Chef::Config[:rubygems_url]
-              cookbook_gems.each do |args|
-                gem(*args)
+            Dir.mktmpdir("chef-gem-bundle") do |dir|
+              File.open("#{dir}/Gemfile", "w+") do |tf|
+                tf.puts "source '#{Chef::Config[:rubygems_url]}'"
+                cookbook_gems.each do |args|
+                  tf.puts "gem #{args.map { |i| "'#{i}'" }.join(' ')}"
```

Comment:
Are we planning support stuff like `git` sources? I think that would have worked with the prior code but not this.